lib/index.mjs

```diff
@@ -0,0 +1,35 @@
+let createRequire = () => {
+  throw new Error(
+    "createRequire is not supported in browser of test environment.",
+  );
+};
+
+try {
+  createRequire = await import("module").then((m) => m.createRequire);
+} catch {}
+
+export default function (filename, opts) {
+  const isBrowser =
+    typeof window !== "undefined" &&
+    typeof document !== "undefined" &&
+    typeof navigator !== "undefined";
+  const isVitest =
+    typeof process !== "undefined" && process.env && process.env.VITEST;
```

Comment:
We can use `std-env` also would `isTest` be enough to make implementation generic?

Reply:
Not sure about other testing env, but generally we do this cause Vitest is based on Vite, let me test a bit with Jest at least. Maybe also for Bun?

Reply:
Yeah. In general, if we can do it runtime-agnostic even if aiming for vitest for now, i would prefer that. If vite/vitest has special behavior, would be happy to also add specific condition (ideally flag from std-env) -- if we have to.

Reply:
Jest seems to neve pick the ESM version, so yeah it should be fine I guess? Updated.